controllers/workspace/devworkspace_controller.go

```diff
@@ -491,8 +491,10 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	reconcileStatus.setConditionTrue(conditions.DeploymentReady, "DevWorkspace deployment ready")
 
 	serverReady, serverStatusCode, err := checkServerStatus(clusterWorkspace)
-	if err != nil {
-		return reconcile.Result{}, err
+	if shouldReturn, reconcileResult, reconcileErr := r.checkDWError(workspace, err, "Error checking server status", metrics.ReasonInfrastructureFailure, reqLogger, &reconcileStatus); shouldReturn {
+		reqLogger.Info("Waiting for DevWorkspace to be ready")
+		reconcileStatus.setConditionFalse(conditions.DeploymentReady, "Waiting for DevWorkspace to be ready") // No sure of the conditions.DeploymentReady is the right one
```

Comment:
`dw.DevWorkspaceReady` may be a more appropriate condition (we use it when the health check fails)